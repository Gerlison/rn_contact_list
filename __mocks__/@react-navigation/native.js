
const RNN = require('@react-navigation/native')
let listeners = {}
const setOptions = jest.fn()
const navigate = jest.fn()

const navigation = {
  setOptions,
  navigate,
  addListener: jest.fn((name, l) => (listeners[name] = l)),
  getListener: (name) => listeners[name],
  triggerListener: (name, ...params) => listeners[name](...params),
  resetListeners: () => {
    listeners = {}
  },
}

const useNavigation = () => navigation
let params = {}
const useRoute = jest.fn(() => ({
  params: params,
}));

module.exports = {
  ...RNN,
  useNavigation,
  useRoute,
  setParams: (p) => (params = { ...params, ...p }),
}
