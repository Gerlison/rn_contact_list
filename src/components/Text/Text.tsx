import React, { memo } from 'react';
import { TextProps } from 'react-native';
import styled, { css } from 'styled-components/native';

type FontFamily = 'Poppins-Regular' | 'Poppins-Medium' | 'Poppins-SemiBold';

interface DefaultTextProps {
  family: FontFamily;
  size: number;
  color: string;
}

export interface Props extends TextProps {
  variant?: FontFamily;
  size?: number;
  color?: string;
  children: React.ReactNode;
}

const Text = ({ children, variant, size, color, ...props }: Props): JSX.Element => (
  <S.DefaultText
    {...props}
    family={variant || 'Poppins-Regular'}
    size={size || 16}
    color={color || '#2f2f2f'}
  >
    {children}
  </S.DefaultText>
);

export const S = {
  DefaultText: styled.Text<DefaultTextProps>`
    ${({ family, size, color }) => css`
      font-family: ${family};
      font-size: ${size}px;
      color: ${color};
    `}
  `,
};

export default memo(Text);
