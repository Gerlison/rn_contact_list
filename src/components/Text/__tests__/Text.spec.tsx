import React from 'react';
import { render } from '@testing-library/react-native';
import 'jest-styled-components';
import 'jest-styled-components/native';

import Text, { Props as TextProps } from '../Text';

describe('Text', () => {
  it('SHOULD match snapshot', () => {
    const sut = render(<Text>foo</Text>).toJSON();
    expect(sut).toMatchSnapshot();
  });

  it.each<[string, TextProps]>([
    [
      'variant',
      {
        variant: 'Poppins-Medium',
      },
    ],
    [
      'size',
      {
        size: 24,
      },
    ],
    [
      'color',
      {
        color: '#F0F0F0',
      },
    ],
  ])('SHOULD match %s style', (_, props) => {
    const sut = render(<Text {...props}>foo</Text>).toJSON();
    expect(sut).toHaveStyleRule('%s', props['%s']);
  });
});
