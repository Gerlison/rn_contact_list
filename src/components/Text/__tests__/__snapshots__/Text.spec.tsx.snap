// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text SHOULD match snapshot 1`] = `
<Text
  color="#2f2f2f"
  family="Poppins-Regular"
  size={16}
  style={
    Array [
      Object {
        "color": "#2f2f2f",
        "fontFamily": "Poppins-Regular",
        "fontSize": 16,
      },
    ]
  }
>
  foo
</Text>
`;
